### PR TITLE
STUN: several improvements

### PR DESCRIPTION
**Pull request by IvanNardi**
Add detection over TCP and fix detection over IPv6.
Rename some variables since Stun dissector is no more "udp-centric".
Stun dissector should always classified the flow as `STUN` or `STUN/Something`.
Don't touch `flow->guessed_host_protocol_id` field, which should be always be related to "ip-classification" only.

### CODE REVIEW

**Review by utoni — COMMENTED** [review]
`mergecap` the `stun*.pcap` files may be a good idea

**Comment by IvanNardi** [comment]
> `mergecap` the `stun*.pcap` files may be a good idea

I merged most of the traces. I kept `stun_signal.pcapng` since it contains a lot of flows correlated among them (basically, it is a single Signal voice call)

File: src/lib/protocols/stun.c [hunk 1]
```diff
@@ -118,43 +118,11 @@ typedef enum {
...
-  31.13.86.5
-*/
-
-static int is_messenger_ip_address(u_int32_t host) {
-  if(host == 0x1F0D5605 /* 31.13.86.5 */)
-    return(1);
-  else
-    return(0);
-}
-
```
Comment (utoni) [inline src/lib/protocols/stun.c:151]: no idea what the code had to do here

**Review by utoni — APPROVED** [review]
good job, a simplification worthy of support

**Pull request merged**